# Тести для малих вибірок та пропорцій



**Завдання 1**. E-commerce компанія після редизайну сайту підозрює, що **середній час до покупки (time-to-purchase)** користувачів **збільшився**.
Історично середній час від першого заходу на сайт до покупки становив $\mu_0 = 123.7$ хвилин.

Команда аналітиків випадково обрала дані **7 покупців після редизайну**:
`128, 135, 121, 142, 126, 151, 123`

З допомогою наявних даних зʼясуйте, чи збільшився середній час покупки після редизайну.

Для цього виконайте наступні 6 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть нульову та альтернативну гіпотези і визначте тип тесту.
2. Обчисліть вибіркові статистики: $\bar x$, $s$, $n$.
3. Оберіть тип тесту та виконайте його (знайдіть критичне значення тесту, статистику тесту та р-значення) будь-яким пасуючим способом, щоб перевірити гіпотезу на рівні значущості $\alpha = 0.10$.
4. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$ використовуючи p-value.
5. Напишіть висновок: чи справді редизайн сайту подовжив час до покупки?
6. Чи зміниться ваше рішення при зміні рівня значущості на $\alpha = 0.05$.


H0 : середній час до покупки досі 123.7 хв

Ha : середній час > 123.7

правосторонній т-тест

In [1]:
import numpy as np
from scipy import stats

# Дані з умови
times = np.array([128, 135, 121, 142, 126, 151, 123])
mu0 = 123.7     # середній час
alpha_10 = 0.10
alpha_05 = 0.05

# Крок 1. Обчислимо вибіркову статистику
x_bar = np.mean(times)
s = np.std(times, ddof=1)   # стандартне відхилення з n-1
n = len(times)

# Крок 2. Обчислимо t-статистику вручну
t_stat = (x_bar - mu0) / (s / np.sqrt(n))

# Крок 3. Критичне значення
df = n - 1
t_critical_10 = stats.t.ppf(1 - alpha_10, df=df)
t_critical_05 = stats.t.ppf(1 - alpha_05, df=df)

# Крок 4. p-value
p_value = 1 - stats.t.cdf(t_stat, df=df)

In [2]:
print(f"\nt-статистика: {t_stat:.2f}")
print(f"Критичне t (alpha=0.10): {t_critical_10:.2f}")
print(f"Критичне t (alpha=0.05): {t_critical_05:.2f}")
print(f"p-value: {p_value:.3f}")

# Крок 5. Прийняття рішень
if p_value < alpha_10:
    print("\nНа рівні значущості 0.10: відхиляємо H0 → час до покупки збільшився.")
else:
    print("\nНа рівні значущості 0.10: не відхиляємо H0.")

if p_value < alpha_05:
    print("На рівні значущості 0.05: відхиляємо H0 → час до покупки збільшився.")
else:
    print("На рівні значущості 0.05: не відхиляємо H0.")


t-статистика: 2.07
Критичне t (alpha=0.10): 1.44
Критичне t (alpha=0.05): 1.94
p-value: 0.042

На рівні значущості 0.10: відхиляємо H0 → час до покупки збільшився.
На рівні значущості 0.05: відхиляємо H0 → час до покупки збільшився.


Cправді редизайн сайту подовжив час до покупки.

**Завдання 2.**

До спеціальної рекламної кампанії **23%** дорослих упізнавали логотип компанії. Після завершення кампанії відділ маркетингу провів опитування: з **1200** випадково відібраних дорослих **311** упізнали логотип.

Перевірте на рівні значущості **3%** ($\alpha=0.03$), чи дають ці дані достатні підстави стверджувати, що **тепер більше ніж 23%** дорослих упізнають логотип компанії. Для розвʼязку використовуйте бібліотеку `statsmodels`.

Зробіть висновок, чи зросла впізнаваність логотипу.

Додатково, обчисліть довірчий інтревал на заданому рівні значущості і проінтерпретуйте текстом - як він додатково пояснює прийняте нами рішення?

In [3]:
import statsmodels.api as sm
from statsmodels.stats.proportion import proportions_ztest, proportion_confint

count = 311   # кількість "успіхів"
nobs = 1200    # обсяг вибірки
p0 = 0.23      # гіпотетична пропорція (H0)

# z-тест для пропорції
stat, pval = proportions_ztest(count, nobs, p0, alternative='larger')
confint = proportion_confint(count, nobs, alpha=0.03, method='normal')

print("🔹 Тест для однієї пропорції")
print(f"Z-статистика = {stat:.3f}")
print(f"p-value = {pval:.4f}")
print(f"97% довірчий інтервал: {confint}")

🔹 Тест для однієї пропорції
Z-статистика = 2.306
p-value = 0.0106
97% довірчий інтервал: (0.23171700302179205, 0.28661633031154127)


In [4]:
lower, upper = confint
if lower > p0:
    print("Нижня межа CI > 0.23 → підтверджує, що впізнаваність зросла.")
else:
    print("Нижня межа CI ≤ 0.23 → дані не дають повної впевненості.")

Нижня межа CI > 0.23 → підтверджує, що впізнаваність зросла.


Якщо p-value < 0.03 → достатньо доказів, що впізнаваність зросла.
Довірчий інтервал покаже, що нижня межа > 0.23 → підтверджує зростання.


Ми 97% впевнені, що істинна частка дорослих, які впізнають логотип після кампанії, знаходиться між 23.2% і 28.7%.
Нижня межа > 0.23 → підтверджує висновок тесту.